js/views/message_view.js

```diff
@@ -498,18 +510,21 @@
       // If we can't verify a signal account locally, we'll go to the Signal Server.
       if (number && !hasLocalSignalAccount) {
         // eslint-disable-next-line more/no-then
-        window.textsecure.messaging
+        this.checkingProfile = window.textsecure.messaging
           .getProfile(number)
           .then(() => {
+            this.contactHasSignalAccount = true;
+
             if (!this.contactView) {
               return;
             }
-
             this.contactView.update(getProps({ hasSignalAccount: true }));
           })
           .catch(() => {
             // No account available, or network connectivity problem
           });
+      } else {
+        this.checkingProfile = Promise.resolve();
       }
```

Comment:
This `render` function has gotten quite long. Would it be possible to extract the part that checks the profile and abstract that it’s memoized?